### ISSUE TITLE
OOPS src/heap.c 472

### ISSUE THREAD

**Issue by flexoron** [trealla-prolog/trealla#155]
v2.14.23
```
?- (((-)-X)=(X-(-)))=Y, Y=X, call(X).
   false.

?- (((-)-X)=(X-(-)))=Y, Y=X, X.
*** OOPS src/heap.c 472
   false.
```

**Comment by flexoron** [comment]
v2.14.24
```
?- ([a,X]=[X,[Y|Z]])=Z, call(Z).
   false. % correct

?- ([a,X]=[X,[Y|Z]])=Z, Z.
*** OOPS src/heap.c 472
   false.
```
```
'True' variant of the example:

?- ([[a|_],X]=[X,[Y|Z]])=Z,call(Z).
   X = [a|_A], Y = a, Z = ([[a|_A],X]=[X,[Y|Z]]).

?- ([[a|_],X]=[X,[Y|Z]])=Z,Z.
*** OOPS src/heap.c 472
   false.
?- 
```

**Comment by flexoron** [comment]
Fixed with v2.14.25. Thanks a lot!
